feature_tests/kotlin/somelib/src/main/kotlin/dev/gigapixel/somelib/Float64Vec.kt

```diff
@@ -0,0 +1,191 @@
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newI16(v: ShortArray): Float64Vec {
+        
+            val (vMem, vSlice) = PrimitiveArrayTools.native(v)
+            
+            val returnVal = lib.Float64Vec_new_i16(vSlice);
+        
+            val selfEdges: List<Any> = listOf()
+            val handle = returnVal
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newU16(v: UShortArray): Float64Vec {
+        
+            val (vMem, vSlice) = PrimitiveArrayTools.native(v)
+            
+            val returnVal = lib.Float64Vec_new_u16(vSlice);
+        
+            val selfEdges: List<Any> = listOf()
+            val handle = returnVal
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newIsize(v: LongArray): Float64Vec {
+        
+            val (vMem, vSlice) = PrimitiveArrayTools.native(v)
+            
+            val returnVal = lib.Float64Vec_new_isize(vSlice);
+        
+            val selfEdges: List<Any> = listOf()
+            val handle = returnVal
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newUsize(v: LongArray): Float64Vec {
+        
+            val (vMem, vSlice) = PrimitiveArrayTools.native(v)
+            
+            val returnVal = lib.Float64Vec_new_usize(vSlice);
+        
+            val selfEdges: List<Any> = listOf()
+            val handle = returnVal
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newF64BeBytes(v: ByteArray): Float64Vec {
+        
+            val (vMem, vSlice) = PrimitiveArrayTools.native(v)
+            
+            val returnVal = lib.Float64Vec_new_f64_be_bytes(vSlice);
+        
+            val selfEdges: List<Any> = listOf()
+            val handle = returnVal
+            val returnOpaque = Float64Vec(handle, selfEdges)
+            CLEANER.register(returnOpaque, Float64Vec.Float64VecCleaner(handle, Float64Vec.lib));
+            vMem.close()
+            return returnOpaque
+        
+        }
+        fun newFromOwned(v: OwnedSlice<F64>): Float64Vec {
```

Comment:
An owned slice should be a `DoubleArray` as well. In GC languages, owned and borrowed slices look the same, because they both always need to be reallocated into the Rust heap.

Reply:
Okay, owned slices are now just converted to kotlin types and the corresponding native memory is immediately freed.